tests/post_training/test_quantize_conformance.py

```diff
@@ -0,0 +1,334 @@
+        return float(fps), cmd_output
+
+    return None, cmd_output
+
+
+def benchmark_performance(model_path, model_name):
+    """
+    Receives the OpenVINO IR model and runs benchmark tool for it
+    """
+
+    model_perf = NOT_AVAILABLE_MESSAGE
+
+    try:
+        model_perf, bench_output = run_benchmark(model_path)
+
+        if model_perf is None:
+            logging.info(
+                f"Cannot measure performance for the model: {model_name}\nDetails: {bench_output}\n"
+            )
+            model_perf = NOT_AVAILABLE_MESSAGE
+    except BaseException as error:
+        logging.error(f"Error when becnhmarking the model: model_name Details: {error}")
+
+    return model_perf
+
+
+def validate_accuracy(model_path, val_loader):
+    dataset_size = len(val_loader)
+    predictions = [0] * dataset_size
+    references = [-1] * dataset_size
+
+    core = ov.Core()
+    ov_model = core.read_model(model_path)
+    compiled_model = core.compile_model(ov_model)
+
+    jobs = int(os.environ.get("NUM_VAL_THREADS", DEFAULT_VAL_THREADS))
+    infer_queue = ov.AsyncInferQueue(compiled_model, jobs)
+
+    def process_result(request, userdata):
+        output_data = request.get_output_tensor().data
+        predicted_label = np.argmax(output_data, axis=1)
+        predictions[userdata] = [predicted_label]
+
+    infer_queue.set_callback(process_result)
+
+    for i, (images, target) in tqdm(enumerate(val_loader)):
+        # W/A for memory leaks when using torch DataLoader and OpenVINO
+        image_copies = copy.deepcopy(images.numpy())
+        infer_queue.start_async(image_copies, userdata=i)
+        references[i] = target
+
+    infer_queue.wait_all()
+    predictions = np.concatenate(predictions, axis=0)
+    references = np.concatenate(references, axis=0)
+
+    return accuracy_score(predictions, references)
+
+
+def benchmark_torch_model(model, dataloader, model_name, output_path):
+    data_sample, _ = next(iter(dataloader))
+    # Dump model
+    onnx_path = Path(output_path) / (model_name + ".onnx")
+    export_to_onnx(model, onnx_path, data_sample)
+    ov_path = Path(output_path) / (model_name + ".xml")
+    export_to_ir(onnx_path, output_path, model_name)
+
+    # Benchmark performance
+    performance = benchmark_performance(ov_path, model_name)
+    # Validate accuracy
+    accuracy = validate_accuracy(ov_path, dataloader)
+    return performance, accuracy
+
+
+def benchmark_onnx_model(model, dataloader, model_name, output_path):
+    # Dump model
+    onnx_path = Path(output_path) / (model_name + ".onnx")
+    onnx.save(model, onnx_path)
+    ov_path = Path(output_path) / (model_name + ".xml")
+    export_to_ir(onnx_path, output_path, model_name)
+
+    # Benchmark performance
+    performance = benchmark_performance(ov_path, model_name)
+    # Validate accuracy
+    accuracy = validate_accuracy(ov_path, dataloader)
+    return performance, accuracy
+
+
+def benchmark_ov_model(model, dataloader, model_name, output_path):
```

Comment:
All of the "benchmark" functions measure both accuracy and performance. I don't think that this is flexible. This could be separated by test case parametrization.

Furthermore, without [special host preparation](https://easyperf.net/blog/2019/08/02/Perf-measurement-environment-on-Linux) both in and out of this test code the performance measurements might not be consistent. It might not be possible to compare the perf results across runs and even across the frameworks in the same run. Complete instructions on how to setup the host state for correct performance measurements, or a reference to a script should be included in the README.

Reply:
Won't fix as a minor.